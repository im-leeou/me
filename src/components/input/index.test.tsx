import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import { Input } from '.';

vi.mock('uuid', () => ({
  v4: () => 'I am mocked'
}));

describe('components input', () => {
  it('renders correctly', () => {
    const result = render(<Input label='name' placeholder='Sang Nguyen' />);

    expect(result).toMatchSnapshot();
  });

  it('renders with error message', () => {
    const result = render(
      <Input label='name' placeholder='Sang Nguyen' errMsg='This is required' />
    );

    const { getByText } = result;
    expect(getByText('This is required')).not.toBeNull();
  });

  it('renders with toggle view password', () => {
    render(
      <Input label='password' type='password' data-testid='input-password' />
    );

    const toggleViewPassword = screen.getByTestId('toggle-view-password');
    expect(toggleViewPassword).not.toBeNull();

    fireEvent.click(toggleViewPassword);
    expect(screen.getByTestId('input-password').getAttribute('type')).toBe(
      'text'
    );
  });
});
