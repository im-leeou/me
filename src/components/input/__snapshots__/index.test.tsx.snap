// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components-input > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col"
      >
        <label
          class="text-gray-secondary mb-2.5 w-fit text-base"
          for="I am mocked"
        >
          _
          name
          :
        </label>
        <div>
          <div
            class="relative w-full"
          >
            <input
              class="shadow-border-line placeholder-gray-subtle w-full rounded-2 bg-black-strong px-4 py-2.5 text-base text-white outline-none transition-all duration-200 ease-in-out focus:ring-2 focus:ring-line/20 focus:ring-offset-1 focus:ring-offset-line/10"
              id="I am mocked"
              placeholder="Sang Nguyen"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col"
    >
      <label
        class="text-gray-secondary mb-2.5 w-fit text-base"
        for="I am mocked"
      >
        _
        name
        :
      </label>
      <div>
        <div
          class="relative w-full"
        >
          <input
            class="shadow-border-line placeholder-gray-subtle w-full rounded-2 bg-black-strong px-4 py-2.5 text-base text-white outline-none transition-all duration-200 ease-in-out focus:ring-2 focus:ring-line/20 focus:ring-offset-1 focus:ring-offset-line/10"
            id="I am mocked"
            placeholder="Sang Nguyen"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
